File: cozumler/not-sistemi-sorusu/javascript/not-sistemi-sorusu.js
const scoreToResultCode = inputScore => {
  const scores = [90, 80, 70, 65, 60, 50, 30, 0]
  const codes = ['AA', 'BA', 'BB', 'CB', 'CC', 'DD', 'FD', 'FF']
  let code

  scores.every((score, idx) => {
    if (inputScore >= score) code = codes[idx]
    return !code
  })

  return code
}

// Test
const testMapping = { 30: 'FD', 90: 'AA', 66: 'CB' }
Object.keys(testMapping).forEach(score => 
  console.assert(scoreToResultCode(score) === testMapping[score])
)

File: cozumler/not-sistemi-sorusu/go/not-sistemi-sorusu.go
package main

var scores = []int{90, 80, 70, 65, 60, 50, 30, 0}
var codes = []string{"AA", "BA", "BB", "CB", "CC", "DD", "FD", "FF"}

func scoreToResultCode(score int) (result string) {
	for idx, _score := range scores {
		if score >= _score {
			result = codes[idx]
			break
		}
	}
	
	return 
}

func test(scores []int, codes []string) bool {
	for idx, _score := range scores {
		if scoreToResultCode(_score) != codes[idx] { panic("ERROR") }
	}

	return true
}

func main() {
	codesToTest := []int{30, 90, 66}
	scoresToTest := []string{"FD", "AA", "CB"}
	test(codesToTest, scoresToTest)
}
